browser/components/MarkdownPreview.js

```diff
@@ -249,30 +245,9 @@ export default class MarkdownPreview extends React.Component {
   }
 
   handleContextMenu (event) {
-    // If a contextMenu handler was passed to us, use it instead of the self-defined one -> return
-    if (_.isFunction(this.props.onContextMenu)) {
-      this.props.onContextMenu(event)
-      return
-    }
-    // No contextMenu was passed to us -> execute our own link-opener
-    if (event.target.tagName.toLowerCase() === 'a' && event.target.getAttribute('href')) {
-      const href = event.target.href
-      const isLocalFile = href.startsWith('file:')
-      if (isLocalFile) {
-        const absPath = uri2path(href)
-        try {
-          if (fs.lstatSync(absPath).isFile()) {
-            context.popup([
-              {
-                label: i18n.__('Show in explorer'),
-                click: (e) => shell.showItemInFolder(absPath)
-              }
-            ])
-          }
-        } catch (e) {
-          console.log('Error while evaluating if the file is locally available', e)
-        }
-      }
+    const menu = buildMarkdownPreviewContextMenu(this, event)
+    if (menu != null) {
+      setTimeout(() => menu.popup(remote.getCurrentWindow()), 30)
```

Comment:
Why do you need setTimeout here?

Reply:
Hmm great question. Copied this from the Code editor context menu has the same behaviour. Do we know why it was added in there? Might be worth just removing it in both places?

Reply:
That code was added in this PR: https://github.com/BoostIO/Boostnote/pull/2338. @ehhc can you explain why you need setTimeout there?

Reply:
To be honest, i can't remember.
Only thing i might imagine is, that setTimeout forces it to be in a separated thread --> not blocking the main gui...
But: there is a good chance, that that's bullshit...

Does it work without the timeout? If yes, feel free to remove it :)

Reply:
@ZeroX-DG thoughts?

Reply:
You heard from the master himself:
> Does it work without the timeout? If yes, feel free to remove it :)
> @ehhc - 2019

Reply:
I have been hoping that this feature would be merged into the base branch. The fact that the original author can not remember why `setTimeout `was used is in itself a worry. If `setTimeout `is removed, the code may work in whatever preliminary testing that it gets, but may fail in some rare timing case in the future. 30ms is not a long time to wait for a menu to appear. I'd say leave it in and move on.

Reply:
@evanpetridis you're right, it is a worry. unfortunately, i cannot do anything about it anymore. Cannot force my way to remember things, i forgot. Nonetheless, at that time i worked with a lot of already existing code, modified it, deleted it, and added new code. All in all i did a lot of work concerning the attachment management of boostnote. I fear, in that context, it's possible to forget the reasons for some things. Even moren, when it's not even clear whether i added it or reused code, i found at other locations in the code base.
And i have to admit, that i, at that point in time and still, am not the expert concerning the technolgies used in boostnote. I hoped for a good codereview to find potential bugs but there is nothing more, i could do about it..

sorry if that's not what you wannted to hear.

Reply:
@ehhc Hi ehhc, my message wasn't very clear. I think that you did the correct thing leaving that code where you did. Well done for actively contributing to open source software :). My point was that because we don't know why it is there, it's an unnecessary danger to remove it. It has no effect leaving it there but we have no idea what the effect of removing it is. I just hope that if the changes that @nathan-castlehow added get sufficiently commented before the changes get merged into the base branch.

Reply:
@ZeroX-DG  i removed it from the preview context menu and it seems ok. I also removed the cut and paste options from the preview context menu as they aren't relevant.